File: services/GroupService/src/main/java/com/se1853_jv/repository/NoteRepository.java
package com.se1853_jv.repository;

import com.se1853_jv.model.Note;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface NoteRepository extends JpaRepository<Note, String> {
    
    /**
     * Find notes by paper ID
     */
    List<Note> findByPaperId(String paperId);
    
    /**
     * Find notes by paper ID and collection ID
     */
    List<Note> findByPaperIdAndCollectionId(String paperId, String collectionId);
    
    /**
     * Find notes by user ID, paper ID, and collection ID
     */
    List<Note> findByUserIdAndPaperIdAndCollectionId(String userId, String paperId, String collectionId);
    
    /**
     * Find notes by paper ID and user ID (for personal collections)
     */
    List<Note> findByPaperIdAndUserId(String paperId, String userId);
}

File: services/paper-service/src/main/kotlin/com/se1853_jv/repository/NoteRepository.kt
package com.se1853_jv.repository

import com.se1853_jv.model.Note
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

@Repository
interface NoteRepository : MongoRepository<Note, String> {
    
    /**
     * Find notes by paper ID
     */
    fun findByPaperId(paperId: String): List<Note>
    
    /**
     * Find notes by paper ID and collection ID
     */
    fun findByPaperIdAndCollectionId(paperId: String, collectionId: String): List<Note>
    
    /**
     * Find notes by user ID, paper ID, and collection ID
     */
    fun findByUserIdAndPaperIdAndCollectionId(
        userId: String,
        paperId: String,
        collectionId: String
    ): List<Note>
    
    /**
     * Find notes by paper ID and user ID (for personal collections)
     */
    fun findByPaperIdAndUserId(paperId: String, userId: String): List<Note>
}







